crates/telio-traversal/src/session_keeper.rs

```diff
@@ -228,18 +261,28 @@ impl Runtime for State {
         F: Future<Output = BoxAction<Self, std::result::Result<(), Self::Err>>> + Send,
     {
         tokio::select! {
-            Ok((pk, action)) = self.keepalive_actions.select_action() => {
+            Ok((pk, action)) = self.nonbatched_actions.select_action() => {
                 let pk = *pk;
-
-                telio_log_trace!("({}) name: \"{}\", action = actions.select_action()", Self::NAME, pk);
-
                 action(self)
                     .await
                     .map_or_else(|e| {
                         telio_log_warn!("({}) Error sending keepalive to {} node: {}", Self::NAME, pk, e.to_string());
                         Ok(())
                     }, |_| Ok(()))?;
             }
+            batched_actions = self.batched_actions.get_future() => {
+                let batched_actions = batched_actions;
```

Comment:
nit: Is this needed? If so why?

Reply:
Not needed, a leftover from the refactoring :))

Reply:
removed it